dask/tests/test_tokenize.py

```diff
@@ -272,21 +247,82 @@ def test_tokenize_numpy_memmap_no_filename():
 
 
 @pytest.mark.skipif("not np")
-def test_tokenize_numpy_ufunc_consistent():
+def test_tokenize_numpy_ufunc():
     assert check_tokenize(np.sin) == check_tokenize("np.sin")
     assert check_tokenize(np.cos) == check_tokenize("np.cos")
 
-    # Make a ufunc that isn't in the numpy namespace. Similar to
-    # any found in other packages.
-    inc = np.frompyfunc(lambda x: x + 1, 1, 1)
-    check_tokenize(inc, copy=False, deterministic=False)
-
     np_ufunc = np.sin
     np_ufunc2 = np.cos
     assert isinstance(np_ufunc, np.ufunc)
     assert isinstance(np_ufunc2, np.ufunc)
     assert check_tokenize(np_ufunc) != check_tokenize(np_ufunc2)
 
+    # for this we'll need the dask.array equivalent
+    inc = da.ufunc.frompyfunc(lambda x: x + 1, 1, 1)
+    inc2 = da.ufunc.frompyfunc(lambda x: x + 1, 1, 1)
+    inc3 = da.ufunc.frompyfunc(lambda x: x + 2, 1, 1)
+    assert check_tokenize(inc) != check_tokenize(inc2)
+    assert check_tokenize(inc) != check_tokenize(inc3)
+
+
+@pytest.mark.skipif("not np")
+def test_normalize_numpy_ufunc_unserializable():
+    # Make a ufunc that isn't in the numpy namespace and can't be serialized
+    inc = np.frompyfunc(lambda x: x + 1, 1, 1)
+    with dask.config.set({"tokenize.ensure-deterministic": False}):
+        # Not idempotent
+        assert tokenize(inc) != tokenize(inc)
+        # You can call normalize_token directly
+        assert normalize_token(inc) != normalize_token(inc)
+
+    with dask.config.set({"tokenize.ensure-deterministic": True}):
+        with pytest.raises(
+            TokenizationError, match=r"Cannot tokenize.*dask\.array\.ufunc.*instead"
+        ):
+            tokenize(inc)
+
+    # Test env override
+    assert tokenize(inc, ensure_deterministic=False) != tokenize(
+        inc, ensure_deterministic=False
+    )
+    with pytest.raises(TokenizationError, match="Cannot tokenize"):
+        tokenize(inc, ensure_deterministic=True)
+
+    # Test ctx variable cleanup
+    with pytest.raises(LookupError):
+        _ensure_deterministic.get()
```

Comment:
This test is about ufuncs. Asserting on this ctxvar feels odd.

Reply:
The test may fail if I were to tamper with the ctxvar without using the helper function `_maybe_raise_nondeterministic`

Reply:
This feels like it would be useful as a comment.

Reply:
I reworked it.